src/facade/dragonfly_connection.h

```diff
@@ -114,13 +121,18 @@ class Connection : public util::Connection {
   using PipelineMessagePtr = std::unique_ptr<PipelineMessage, MessageDeleter>;
   using PubMessagePtr = std::unique_ptr<PubMessage, MessageDeleter>;
 
+  // Variant wrapper around different message types
   struct MessageHandle {
     size_t UsedMemory() const;  // How much bytes this handle takes up in total.
 
+    // Intrusive messages put themselves at the front of the queue, but only after all other
+    // intrusive ones. Used for quick transfer or control / update messages.
+    bool IsIntrusive() const;
```

Comment:
I think `IsControlMsg()` is a better name, because "intrusiveness" is an implementation detail.

Reply:
I agree, however `MigrationRequestMessage` is a control message, but it's not intrusive 🤔 The only "implementation" detail we expose here is that we have a linear order of messages

Reply:
why the order is important for your logic?

Reply:
because ACL updates and checkpoint need to be applied immediately (after the next command finished), where as the migration request wants to be last explicitly to make sure all commands before it have been processed